src/main/java/ru/ayaz/ChatServer.java

```diff
@@ -0,0 +1,184 @@
+
+        System.out.println(message);
+
+        for (Map.Entry<String, User> entry : usersMap.entrySet()) {
+            String nickname = entry.getKey();
+            User user = entry.getValue();
+
+            if (user.isItMe(sender)) {
+                continue;
+            }
+
+            if (!user.isIgnored(sender)) {
+                PrintWriter pw = outputWriterMap.get(nickname);
+                pw.println(message);
+                pw.flush();
+            }
+
+        }
+
+    }
+
+
+    public class UserHandler implements Runnable {
+
+        private User user;
+        private MessageProcessor processor;
+        private PrintWriter writer;
+        private BufferedReader reader;
+
+
+        public UserHandler(Socket socket) {
+            try {
+                this.reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+                this.writer = new PrintWriter(socket.getOutputStream());
+                this.user = new User();
+                this.processor = new MessageProcessor();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+
+        public void run() {
+
+            String message;
+            String sender;
+
+            welcomeUser();
+
+            try {
+
+                user.setNickname(listenForMessage());
+                addUser();
+                addWriter();
+
+                sender = user.getNickname();
+
+                sayEveryoneUserConnected(sender);
+
+                while ((message = listenForMessage()) != null) {
+                    if (!processor.isCommand(message)) {
+                        sendMessageToEveryone(sender + ": " + message, sender);
+                    } else {
+                        executeCommand(message);
+                    }
+                }
+
+                sayEveryoneUserLeft(sender);
+
+                Thread.currentThread().interrupt();
+
+            } catch (InvalidNicknameException e) {
+                writer.println(e.toString());
+                writer.flush();
+                writer.close();
+                try {
+                    reader.close();
+                } catch (IOException e1) {
+                    e1.printStackTrace();
+                }
+            } catch (IOException e) {
+                sayEveryoneUserLeft(user.getNickname());
+            }
+        }
+
+
+        private void welcomeUser() {
+            writer.println("Welcome to Chat");
+            writer.println("Please enter your name:\n");
+            writer.flush();
+        }
+
+
+        private void addUser() {
+            usersMap.put(user.getNickname(), user);
+        }
+
+
+        private void addWriter() {
+            outputWriterMap.put(user.getNickname(), writer);
+        }
+
+
+        private String listenForMessage() throws IOException {
+            return reader.readLine();
+        }
+
+
+        private void sayEveryoneUserConnected(String connectedUser) {
```

Comment:
&#x1F34E; **Issue**
sayEveryoneUserConnected ->
sayEveryoneThatUserHasConnected
If its not too long, don't spare words for naming )